### PR TITLE
Planck: Layout Macro Refactor

### PR DESCRIPTION
**Pull request by noroadsleft**
Currently certain layout macros are specific to either AVR or ARM when used in the QMK Configurator. If an AVR-specific macro is used for a Planck rev. 6, or an ARM-specific macro on a rev. 5 or earlier, the user receives a compile error.

Hoping to unify the macro names for both architectures.

Feedback desired and appreciated. I know some of the keymaps fail with errors, but I consider that out-of-scope for this PR, but it can be discussed.

### CODE REVIEW

**Comment by drashna** [comment]
This supersedes #4329

File: keyboards/planck/planck.h [hunk 1]
```diff
@@ -135,12 +117,29 @@
...
+  KC_##k00, KC_##k01, KC_##k02, KC_##k03, KC_##k04, KC_##k05, KC_##k06, KC_##k07, KC_##k08, KC_##k09, KC_##k0a, KC_##k0b, \
+  KC_##k10, KC_##k11, KC_##k12, KC_##k13, KC_##k14, KC_##k15, KC_##k16, KC_##k17, KC_##k18, KC_##k19, KC_##k1a, KC_##k1b, \
+  KC_##k20, KC_##k21, KC_##k22, KC_##k23, KC_##k24, KC_##k25, KC_##k26, KC_##k27, KC_##k28, KC_##k29, KC_##k2a, KC_##k2b, \
+  KC_##k30, KC_##k31, KC_##k32, KC_##k33, KC_##k34, KC_##k35, KC_##k36, KC_##k37, KC_##k38, KC_##k39, KC_##k3a, KC_##k3b \
+)
+
+#define KEYMAP LAYOUT_ortho_4x12
+#define LAYOUT_planck_mit  LAYOUT_planck_1x2uC
+#define LAYOUT_planck_grid LAYOUT_ortho_4x12
+#define KC_LAYOUT_ortho_4x12 LAYOUT_kc
```
Comment (drashna) [inline keyboards/planck/planck.h]: Shouldn't this be: 
```suggestion
#define LAYOUT_kc_ortho_4x12 LAYOUT_kc
```

Comment (noroadsleft) [reply keyboards/planck/planck.h]: You might be right. I haven't gone through the keymaps to see what's actually used.

**Comment by noroadsleft** [comment]
Wow @drashna, I totally forgot that you had that PR already. I even *commented.* :scream:

File: keyboards/planck/planck.h (hunk 1 above)
Comment (noroadsleft) [reply keyboards/planck/planck.h]: Okay, looks like `planck:bakingpy` via ortho_4x12 community layout uses `LAYOUT_kc_ortho_4x12`, and `planck:corvec` (native Planck keymap) uses `KC_KEYMAP`.

File: keyboards/planck/planck.h [hunk 2]
```diff
@@ -135,12 +117,29 @@
...
-#define LAYOUT_ortho_4x12 LAYOUT_planck_grid
-#define KC_LAYOUT_ortho_4x12 KC_KEYMAP
+  }
 
 #endif
 
+// all Planck keyboards
+
+// Used to create a keymap using only KC_ prefixed keys
+#define LAYOUT_kc( \
```
Comment (drashna) [inline keyboards/planck/planck.h]:
```suggestion
#define LAYOUT_kc_ortho_4x12( \
```

File: keyboards/planck/planck.h [hunk 3]
```diff
@@ -135,12 +117,29 @@
...
+  KC_##k00, KC_##k01, KC_##k02, KC_##k03, KC_##k04, KC_##k05, KC_##k06, KC_##k07, KC_##k08, KC_##k09, KC_##k0a, KC_##k0b, \
+  KC_##k10, KC_##k11, KC_##k12, KC_##k13, KC_##k14, KC_##k15, KC_##k16, KC_##k17, KC_##k18, KC_##k19, KC_##k1a, KC_##k1b, \
+  KC_##k20, KC_##k21, KC_##k22, KC_##k23, KC_##k24, KC_##k25, KC_##k26, KC_##k27, KC_##k28, KC_##k29, KC_##k2a, KC_##k2b, \
+  KC_##k30, KC_##k31, KC_##k32, KC_##k33, KC_##k34, KC_##k35, KC_##k36, KC_##k37, KC_##k38, KC_##k39, KC_##k3a, KC_##k3b \
+)
+
+#define KEYMAP LAYOUT_ortho_4x12
+#define LAYOUT_planck_mit  LAYOUT_planck_1x2uC
+#define LAYOUT_planck_grid LAYOUT_ortho_4x12
+#define LAYOUT_kc_ortho_4x12 LAYOUT_kc
```
Comment (drashna) [inline keyboards/planck/planck.h]: Looks like a couple keymaps still use KC_KEYMAP :(

```suggestion
#define LAYOUT_kc_ortho_4x12 LAYOUT_kc
#define KC_KEYMAP LAYOUT_kc
```

Comment (noroadsleft) [reply keyboards/planck/planck.h]: Long-term, I'd like to update all the keymaps so the ones that use `KC_`-prepending macro at least all use the same name for it.

**Comment by noroadsleft** [comment]
I just realized two things:

1. Git via MinGW x64 isn't signing my commits.
2. If I use your suggestions, you're the commit author, even though I'm the one writing the commit message. :man_shrugging:

**Comment by drashna** [comment]
@noroadsleft yup.  It can though.  If you have gpg installed, you can configure it to do so.
https://help.github.com/articles/telling-git-about-your-signing-key/

And yeah, it's weird how the suggestions work....

**Comment by noroadsleft** [comment]
*bump* Does this need anything else? All the Travis CI errors seem to be hex files that are too big for AVR Plancks, which is an issue I consider out-of-scope here.

**Comment by noroadsleft** [comment]
@drashna Updated. Could you check out the 1x2uC macro and tell me if it's correct? Once all the layout macros are correct I'll check `info.json` and update it if needed.

Also, a lot of the Planck keymaps are using raw matrix instead of layout macros. Travis also isn't checking rev6 compilation because `make planck` defaults to rev5.

File: keyboards/planck/planck.h [hunk 4]
```diff
@@ -9,128 +9,111 @@
...
+  ) \
+  { \
+    { k00,   k01, k02, k03, k04, k05 }, \
+    { k10,   k11, k12, k13, k14, k15 }, \
+    { k20,   k21, k22, k23, k24, k25 }, \
+    { k30,   k31, k32, k39, k3a, k3b }, \
+    { k06,   k07, k08, k09, k0a, k0b }, \
+    { k16,   k17, k18, k19, k1a, k1b }, \
+    { k26,   k27, k28, k29, k2a, k2b }, \
+    { KC_NO, k37, k38, k33, k34, k35 } \
```
Comment (drashna) [inline keyboards/planck/planck.h]: I had to change this to get it to work: 
```suggestion
    { k35, k37, k38, k33, k34, KC_NO} \
```

Comment (noroadsleft) [reply keyboards/planck/planck.h]: TY sir.

**Comment by noroadsleft** [comment]
I think we're done here.

**Comment by drashna** [comment]
@jackhumbert

**Comment by karma0** [comment]
Should rev6 users be using the noroadsleft:fix/planck branch?

**Pull request merged**